packages/dd-trace/src/span_pointers.js

```diff
@@ -0,0 +1,33 @@
+const crypto = require('crypto')
+
+const SPAN_LINK_KIND = 'span-pointer'
+const S3_PTR_KIND = 'aws.s3.object'
+
+const SPAN_POINTER_DIRECTION = Object.freeze({
+  UPSTREAM: 'u',
+  DOWNSTREAM: 'd'
+})
+
+/**
+ * Generates a unique hash for an S3 object using its bucket name, key, and ETag
+ * https://github.com/DataDog/dd-span-pointer-rules/blob/main/AWS/S3/Object/README.md
+ * @param {string} bucketName - The name of the S3 bucket containing the object
+ * @param {string} objectKey - The full path/key of the object in the bucket
+ * @param {string} eTag - The ETag value from S3, which may be wrapped in quotes
+ * @returns {string} A hash uniquely identifying the S3 request.
+ */
+function generateS3PointerHash (bucketName, objectKey, eTag) {
```

Comment:
a little weird that we have an aws s3-specifc funciton here in a file that seems to be general span pointers code?

Reply:
I was going to put all the hashing functions in this file. And once I get to Dynamo, I can pull out any reused parts into this file or use a switch statement inside a single 'generateHash' function